quartz/components/scripts/search.inline.ts

```diff
@@ -164,12 +268,183 @@ async function setupSearch(searchElement: Element, currentSlug: FullSlug, data:
   const searchLayout = searchElement.querySelector(".search-layout") as HTMLElement
   if (!searchLayout) return
 
+  const searchSpace = searchElement?.querySelector(".search-space") as HTMLFormElement
+  if (!searchSpace) return
+
+  // Create semantic search progress bar
+  const progressBar = document.createElement("div")
+  progressBar.className = "semantic-search-progress"
+  progressBar.style.cssText = `
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    height: 2px;
+    width: 0;
+    background: var(--secondary);
+    transition: width 0.3s ease, opacity 0.3s ease;
+    opacity: 0;
+    z-index: 9999;
+  `
+  searchBar.parentElement?.appendChild(progressBar)
+
+  const startSemanticProgress = () => {
+    progressBar.style.opacity = "1"
+    progressBar.style.width = "0"
+    setTimeout(() => {
+      progressBar.style.width = "100%"
+    }, 10)
+  }
+
+  const completeSemanticProgress = () => {
+    progressBar.style.opacity = "0"
+    setTimeout(() => {
+      progressBar.style.width = "0"
+    }, 300)
+  }
+
+  const resetProgressBar = () => {
+    progressBar.style.opacity = "0"
+    progressBar.style.width = "0"
+  }
+
   const idDataMap = Object.keys(data) as FullSlug[]
+  const slugToIndex = new Map<FullSlug, number>()
+  idDataMap.forEach((slug, idx) => slugToIndex.set(slug, idx))
+  const modeToggle = searchSpace.querySelector(".search-mode-toggle") as HTMLDivElement | null
```

Comment:
Does this work when multiple search components are present? (E.g. different position for mobile/desktop)